pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/client/Auth0ManagementClient.java

```diff
@@ -527,31 +529,55 @@ private ApiResult<String, APIException> tryPasswordChangeRequest(PasswordChangeT
         });
     }
 
-    private <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResult<B, E>> callback) {
+    public <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResult<B, E>> callback) {
         ApiResult<B, E> res = null;
         int numTries = 0;
         int maxTries = maxRetries + 1;
+        long waitTotal = 0L;
         while (numTries < maxTries) {
             res = callback.get();
             numTries++;
             if (numTries >= maxTries) {
                 break;
             }
-            if (res.getStatusCode() == 429) {
+            long wait = backoffMillis * numTries + new Random().nextInt(MAX_JITTER_MILLIS);
+            if (res.getError() instanceof RateLimitException && res.getStatusCode() == 429) {
                 log.error(retryMessage, res.getError());
```

Comment:
We probably do not want to log at error level since that will generate a prod alert. That *might* be appropriate if we are unable to handle this situation (exceed maxTries), but just getting the 429 should probably not alert. (You can imagine case where we bombard ourselves with alerts on each retry.)